subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy

```diff
@@ -379,11 +379,18 @@ class TestTaskIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         buildFile << """apply plugin: 'java'
 
             ${mavenCentralRepository()}
+
+            sourceSets {
+                customTest
+            }
+
             dependencies {
-                testImplementation 'junit:junit:${JUnitCoverage.NEWEST}'
+                customTestImplementation 'junit:junit:${JUnitCoverage.NEWEST}'
             }
 
             tasks.create('customTest', Test) {
+                classpath = sourceSets.customTest.runtimeClasspath
+                testClassesDirs = sourceSets.customTest.output.classesDirs
```

Comment:
This breaks the test because we're now not going to find any test classes.  I think we want to move the source into the right path or configure customTest to use test's conventions.

Reply:
Moved the source files